src/main/java/com/exactpro/th2/sailfish/utils/ProtoToIMessageConverter.java

```diff
@@ -171,7 +171,15 @@ public IMessage fromMetadataFilter(MetadataFilter filter, String messageName) {
         IMessage message = messageFactory.createMessage(dictionaryURI, messageName);
         for (Entry<String, SimpleFilter> filterEntry : filter.getPropertyFiltersMap().entrySet()) {
             SimpleFilter propertyFilter = filterEntry.getValue();
-            message.addField(filterEntry.getKey(), toSimpleFilter(propertyFilter.getOperation(), propertyFilter.getValue()));
+            if (propertyFilter.hasSimpleList()) {
+                if (propertyFilter.getOperation() == FilterOperation.IN || propertyFilter.getOperation() == FilterOperation.NOT_IN) {
+                    message.addField(filterEntry.getKey(), new ListContainFilter(propertyFilter.getOperation(), propertyFilter.getSimpleList().getSimpleValuesList()));
+                } else {
+                    throw new IllegalArgumentException(String.format("The operation doesn't match the values {%s}, {%s}", propertyFilter.getOperation(), propertyFilter.getSimpleList()));
+                }
```

Comment:
```suggestion
                if (propertyFilter.getOperation() != FilterOperation.IN && propertyFilter.getOperation() != FilterOperation.NOT_IN) {
                throw new IllegalArgumentException(String.format("The operation doesn't match the values {%s}, {%s}", propertyFilter.getOperation(), propertyFilter.getSimpleList()));
                }
                message.addField(filterEntry.getKey(), new ListContainFilter(propertyFilter.getOperation(), propertyFilter.getSimpleList().getSimpleValuesList()));
```

Reply:
Please, add a test for the new option to `ProtoToIMessageConverterWithoutDictionaryTest` class